packages/ui/src/components/card.tsx

```diff
@@ -1,89 +1,138 @@
-import { forwardRef, HTMLAttributes } from 'react'
+import { Children, forwardRef, HTMLAttributes, ImgHTMLAttributes, isValidElement, ReactNode } from 'react'
 
 import { cn } from '@utils/cn'
 import { cva, type VariantProps } from 'class-variance-authority'
 
-const cardVariants = cva('rounded-lg border bg-cn-background-2 text-cn-foreground-1 shadow', {
+const cardVariants = cva('cn-card', {
   variants: {
-    variant: {
-      default: '',
-      plain: 'border-none bg-transparent shadow-none'
+    size: {
+      sm: 'cn-card-sm',
+      default: 'cn-card-default',
+      md: 'cn-card-md'
+    },
+    orientation: {
+      vertical: 'cn-card-vertical',
+      horizontal: 'cn-card-horizontal'
+    },
+    position: {
+      start: 'cn-card-position-start',
+      end: 'cn-card-position-end'
+    },
+    selected: {
+      true: 'cn-card-selected',
+      false: ''
+    },
+    disabled: {
+      true: 'cn-card-disabled',
+      false: ''
     }
   },
   defaultVariants: {
-    variant: 'default'
+    size: 'default',
+    orientation: 'vertical',
+    position: 'start',
+    selected: false,
+    disabled: false
   }
 })
 
 export interface CardRootProps extends HTMLAttributes<HTMLDivElement>, VariantProps<typeof cardVariants> {
-  width?: 'auto' | 'full' | 'screen' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | string
+  size?: 'sm' | 'default' | 'md'
+  children: ReactNode
 }
 
-const widthVariants = {
-  auto: 'w-auto',
-  full: 'w-full',
-  screen: 'w-screen',
-  fit: 'w-fit',
-  xs: 'w-24',
-  sm: 'w-48',
-  md: 'w-64',
-  lg: 'w-72',
-  xl: 'w-80',
-  '2xl': 'w-96'
-}
-
-const CardRoot = forwardRef<HTMLDivElement, CardRootProps>(({ variant, className, width = 'auto', ...props }, ref) => {
-  const widthClassName = widthVariants[width as keyof typeof widthVariants] || width
-
-  return <div ref={ref} className={cn(cardVariants({ variant }), widthClassName, className)} {...props} />
-})
+const CardRoot = forwardRef<HTMLDivElement, CardRootProps>(
+  (
+    {
+      className,
+      size = 'default',
+      selected = false,
+      disabled = false,
+      orientation = 'vertical',
+      position = 'start',
+      children,
+      ...props
+    },
+    ref
+  ) => {
+    const { imageContent, otherContent } = Children.toArray(children).reduce<{
+      imageContent: ReactNode | null
+      otherContent: ReactNode[]
+    }>(
+      (acc, child) => {
+        if (acc.imageContent === null && isValidElement(child) && child.type === CardImage) {
+          acc.imageContent = child
+        } else {
```

Comment:
Unfortunately this won't discard additional image elements, it'll just display them in the content portion.
```suggestion
        if (isValidElement(child) && child.type === CardImage) {
          if (!!acc.imageContent) acc.imageContent = child
        } else {
```

Reply:
Fixed